### ISSUE TITLE
Twitter User Mentions feed

### ISSUE THREAD

**Issue by skpy** [snarfed/granary#153]
Twitter offers an API endpoint for [user mentions](https://developer.twitter.com/en/docs/tweets/timelines/api-reference/get-statuses-mentions_timeline.html).  It would be grand if I could subscribe to this from Granary.

**Comment by snarfed** [comment]
yes! thanks for filing. i looked at that API endpoint originally, but shied away from it since it only works for the authenticated user, not arbitrary usernames: 

https://github.com/snarfed/granary/blob/master/granary/twitter.py#L415-L418

...but that's a totally reasonable limit to impose.

**Comment by snarfed** [comment]
note to myself: i'd probably add the special case for the authenticated user to `fetch_mentions()`:

https://github.com/snarfed/granary/blob/master/granary/twitter.py#L449

not sure how i'd expose it in the API and/or the web UI yet.

**Comment by skpy** [comment]
Sorry if I'm missing something obvious, but isn't Granary authenticating to Twitter as me already?  Or is it authenticating as your application ID and I'm simply granting that ID read access to my Tweets?

**Comment by snarfed** [comment]
heh, yes. the missing part is the `statuses/mentions_timeline` API call, and the new REST endpoint, UI, etc we'll need for it, not authentication.

**Comment by snarfed** [comment]
[Twitter is now developer-hostile](https://snarfed.org/2023-04-03_so-long-twitter-api-and-thanks-for-all-the-fish), so I don't plan to work on this myself. I'll happily review and merge PR(s) though!
